src/movies.js

```diff
@@ -1,16 +1,68 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(arr) {
+  return arr.map((item) => item.director);
+}
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(arr) {
+  return arr.filter(
+    (item) =>
+      item.director === "Steven Spielberg" && item.genre.includes("Drama")
+  ).length;
+}
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(arr) {
+  const arrRate = arr.map((item) => item.rate);
```

Comment:
When you map over an array which does not have a rate - it returns - undefined. That's why when you reduce it later it returns NaN.